### ISSUE TITLE
Question: How to add tags to all spans?

### ISSUE THREAD

**Issue by jlsalmon** [opentracing-contrib/java-specialagent#437]
Hello,

First of all, thanks for contributing this excellent library, and apologies if this question is better suited to a different forum - please feel free to redirect me if that is the case.

Context: I'm currently using the Jaeger tracing plugin, which uses the concept of a "Process" object to house tags that are common across all spans in a trace (e.g. `service_name`). However, I'm also using a downstream component (Honeycomb) which expects `the service_name` tag to be present on each individual span, and doesn't recognise the "Process" object.

Question: Is it possible to somehow inject a piece of code into the specialagent which can decorate each span (regardless of specific instrumentation plugin) and add some tags before sending?

Looks like #415 might be relevant.

Many thanks in advance!

**Comment by safris** [comment]
Hi @jianshaow, thank you for your question. The short and most immediate answer is: no, there is no built-in mechanism for applying generic tags to all spans. SpecialAgent is merely an engine that automatically installs [Instrumentation Plugins](https://github.com/opentracing-contrib/java-specialagent/#61-instrumentation-plugins) into a runtime. But, like you already found by yourself, #415 is indeed relevant. This is a new feature being developed by @zeitlinger, and we are planning to release it with [v1.5.9](https://github.com/opentracing-contrib/java-specialagent/milestone/6). As of the current moment, there is unfortunately no way for you to achieve what you desire. However, with the completion of #415, this feature _may be_ available. I say _may be_, because the original scope of #415 involved blacklisting or transforming existing tags, fields or logs.

@zeitlinger, what do you think of the use-case that @jlsalmon has outlined?

**Comment by zeitlinger** [comment]
Yes, great to have another use case before it's released. I'll have a look.

**Comment by zeitlinger** [comment]
@safris @jlsalmon I gave it a try with #439 - and hope it makes some sense.

Have a look at the [README](https://github.com/opentracing-contrib/java-specialagent/blob/578846142ddf0c7e019607f3d6203586b2d5c5ea/opentracing-specialagent-util/src/main/java/io/opentracing/contrib/specialagent/tracer/README.md#global-rules) to get an idea.

**Comment by jlsalmon** [comment]
@safris @zeitlinger thanks for the super quick responses!

If I understand correctly, with #439 the following config should add the service name to all spans under the `service_name` key - is that right?

```
{
  "all": [
     "type": "serviceName",
     "output": [
       {
         "type": "tag", 
         "key": "service_name"
       }
     ]
  ]
}
```

**Comment by zeitlinger** [comment]
Correct 😀

**Comment by jlsalmon** [comment]
Great, thanks @zeitlinger.  Looking forward to giving this a spin once it's all glued together!

**Comment by jlsalmon** [comment]
I see this has been shipped in #479 - thanks guys :)

**Comment by safris** [comment]
That's correct @jlsalmon. This will be part of the [v1.6.0](https://github.com/opentracing-contrib/java-specialagent/milestone/7) release this coming weekend.
